### DISCUSSION TITLE
Share Authentication session between multiple NextJS projects using next-auth and turbo-repo

### DISCUSSION THREAD

**Discussion by sumodhnair19** [vercel/turborepo#3247] [Help] [answered]
Hi,

First of all, huge thank you for `turborepo`. It is amazing. 

I would like to run by you a scenario. I have multiple `nextJS` projects. Both of them requires authentication from a user in order to visit certain pages i.e. protected routes. I am using `next-auth.js` with a credentials provider for doing the same. I have a login page which does the authentication process for both the `NextJS` apps.

I wanted to see if the scenario mentioned above is possible via `turborepo` by sharing user's `session-state` between the two `NextJS` projects. 

My use case is: One login page-> user logs in -> protected routes between both projects can now be accessed as `session` will be shared?

I was looking into the examples for `turborepo`. However, I can't seem to find any example for the same? If you could please advise on this scenario. That will be amazing.

Thanks,
Sumodh

**Comment by EarlGeorge** [comment, marked as answer]
personally I wouldn't recommend. it doesn't work that way try to avoid this.

**Reply by sumodhnair19** [reply]
@EarlGeorge Thanks for your response. Could you please explain why it won't work that way? Also, can you please suggest a way to make this scenario work? 

There is a lot of packages in the project which can be shared between the projects. Also, I want to have a one stop login, which once logged in, provides access to all the projects.
